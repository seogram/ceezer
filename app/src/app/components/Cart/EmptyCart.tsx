import React from "react";
import { usePathname } from 'next/navigation'
import { Stack, Typography } from "@mui/material";
import CustomeLink from "../Link/CustomeLink";


const EmptyCart = () => {
  const pathname = usePathname()
  return (
    <Stack direction="column">
      <Typography>
        There is no item in shopping cart.
      </Typography>
      {pathname !=="/" && (
        <CustomeLink href="/">
          <Typography color="primary">
            Back
          </Typography>
        </CustomeLink>
      )}
    </Stack>
  );
}

export default EmptyCart;