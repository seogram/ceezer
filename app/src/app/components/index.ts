export { default as MarketPlace } from "./MarketPlace";
export { default as Loading } from "./Loading/Loading";
export { default as CustomError } from "./Error/Error";
export { default as Header } from "./Header/Header";
export { default as Cart } from "./Cart/Cart";
export { default as CustomeLink } from "./Link/CustomeLink";
export { default as SearchBox } from "./SearchBox/SearchBox";

