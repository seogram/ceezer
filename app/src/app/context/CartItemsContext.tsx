
"use client";

import { createContext, ReactNode } from "react";
import { useState, useEffect } from "react";
import { setLocalStorage, getLocalStorage } from "../utils";
import { CartItem as CartItemType } from "../type";

type CartItemsContext = {
  cartItems: CartItemType[],
  setCartItems: (cartItems: CartItemType[]) => void
}

export const CartItemsContext = createContext<CartItemsContext>({
  cartItems: [],
  setCartItems: () => {},
});

export const CartItemsProvider = ({ children }: { children: ReactNode }) => {

  const [cartItems, setCartItems] = useState<any[]>([]);
  useEffect(() => {
    const currentCartItems = getLocalStorage("cart");
    if (currentCartItems) {
      setCartItems(currentCartItems)
      setLocalStorage("cart", currentCartItems)
    }
  }, [])

  return (
    <CartItemsContext.Provider value={{ cartItems, setCartItems }}>
      {children}
    </CartItemsContext.Provider>
  );
};