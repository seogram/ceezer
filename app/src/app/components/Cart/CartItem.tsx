import Image from "next/image";
import { Typography, Box, Stack, IconButton } from "@mui/material";
import { styled } from '@mui/material/styles';
import { formatCurrency } from "@/app/utils";
import DeleteOutlineIcon from '@mui/icons-material/DeleteOutline';
import { CartItem as CartItemType } from "@/app/type";
import { useCartActions } from "@/app/hooks";

type Props = {
    item: CartItemType;
    addToCart: (clickedItem: CartItemType) => void;
    removeFromCart: (item: CartItemType) => void;
};

const Wrapper = styled(Stack)(() => ({
    padding: "1rem 0",
}));

const InfoWrapper = styled(Box)(() => ({
    textAlign: "left",
    paddingBottom: "1.5rem"
}));



const CartItem = ({ item, removeFromCart }: Props) => {
    const { updateItem } = useCartActions();

    const cartUpdateHandler = (item: CartItemType, action: "DECREASE" | "INCREASE") => {
        
        if (item && action === "INCREASE") {
            const existingVolume = item.volume;
            const { offeredVolume } = item
            if (existingVolume === offeredVolume || existingVolume + 0.5 > offeredVolume) {
                return;
            }
        }

        updateItem(item, action)
    }

    return (
        <Wrapper direction="row" gap={2} >
            <Box sx={{ flex: 1 }}>
                <InfoWrapper>
                    <Typography variant="h6">{item.name}</Typography>
                    <Typography>Price per ton: {formatCurrency(item.pricePerTon)}</Typography>
                    <Typography>Volume(ton): {item.volume}</Typography>
                    <Typography>Total Price: {formatCurrency(item.pricePerTon * item.volume)}</Typography>
                </InfoWrapper>
                <Stack direction="row" spacing={1} alignItems="center">
                    <IconButton
                        data-testid="decrease-volume"
                        size="large"
                        onClick={() => cartUpdateHandler(item, "DECREASE")}
                    >
                        -
                    </IconButton>
                    <Box>
                        {item.volume}
                    </Box>
                    <IconButton
                        data-testid="increase-volume"
                        size="large"
                        onClick={() => cartUpdateHandler(item, "INCREASE")}
                    >
                        +
                    </IconButton>
                    <Box
                        sx={{
                            display: {
                                xs: "flex",
                                sm: "none"
                            }
                        }}
                    >
                        <IconButton
                            data-testid="remove-item"
                            size="large"
                            edge="end"
                            onClick={() => removeFromCart(item)}
                            color="inherit"
                        >
                            <DeleteOutlineIcon />
                        </IconButton>
                    </Box>
                </Stack>
            </Box>
            <Box>
                <Image
                    src={item.image ?? ""}
                    alt={item.name}
                    width={150}
                    height={160}
                />
            </Box>
            <Box
                sx={{
                    display: {
                        xs: "none",
                        sm: "block"
                    }
                }}
            >
                <IconButton
                    size="large"
                    edge="end"
                    onClick={() => removeFromCart(item)}
                    color="inherit"
                >
                    <DeleteOutlineIcon />
                </IconButton>
            </Box>
        </Wrapper>
    );
};

export default CartItem;
