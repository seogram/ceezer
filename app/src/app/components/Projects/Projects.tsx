import React from 'react';
import { Typography, Button, Box, Stack } from '@mui/material';
import { styled } from '@mui/material/styles';
import { useData } from '@/app/hooks';
import ProjectDetail from '../ProjectDetail/ProjectDetail';
import ErrorComponent from '../Error/Error';
import Loading from '../Loading/Loading';
import { useRouter } from "next/navigation";
import type { Projects as ProjectsType } from '@/app/type';

const RootStyle = styled(Stack)({
  gap: '20px',
});

const StyledBox = styled(Box)({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-between'
});

interface Props {
  searchTerm?: string;
}

interface UseProjectDataReturn {
  data: ProjectsType;
  isFetching: boolean;
  isError: boolean;
}

const renderDetails = (projects: ProjectsType) =>
  projects.map((project) => <ProjectDetail projectData={project} key={project.id} />);


const Projects = ({ searchTerm = '' }: Props) => {
  const router = useRouter();
  const { data: projects, isFetching: isProjectFetching, isError } = useData() as UseProjectDataReturn;

  if (isError) return <ErrorComponent />;
  if (isProjectFetching) return <Loading />;

  const filteredProjects = searchTerm
    ? projects.filter((project) =>
      project.name.toLowerCase().includes(searchTerm.toLowerCase()))
    : projects;

  return (
    <RootStyle>
      {searchTerm && (
        <StyledBox>
          <Typography data-testid="result">Showing Projects for: {searchTerm}</Typography>
          <Button name="reset" onClick={() => router.replace('/')}>
            Reset Result
          </Button>
        </StyledBox>
      )}
      <Box>{renderDetails(filteredProjects)}</Box>
    </RootStyle>
  );
};

export default Projects;