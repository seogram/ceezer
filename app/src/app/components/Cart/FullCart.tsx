
import { useRouter } from "next/navigation";
import { Typography, Button, Box, Stack, Divider } from "@mui/material";
import { styled } from '@mui/material/styles';
import { useCartActions } from "@/app/hooks";
import { CartItem as CartItemType } from "@/app/type";
import CartItem from "./CartItem";
import { formatCurrency } from "@/app/utils";
import EmptyCart from "./EmptyCart";
import { memo } from "react";

const StyledBox = styled(Box)(({ theme }) => ({
    padding: theme.spacing(2),
    ...theme.typography.body2,
    textAlign: 'center',
}));

const FullCart = ({ totalCost, cartItems }: { totalCost: number, cartItems: CartItemType[] }) => {

    const router = useRouter();
    const { addItem, removeItem } = useCartActions();

    const handlePayNow = () => router.push("/");

    const hasCartItems = cartItems.length > 0;

    return (
        <StyledBox data-testid="cart">
            {
                hasCartItems ? (
                    <>
                        {cartItems.map((item) => (
                            <CartItem
                                key={item.id}
                                item={item}
                                addToCart={addItem}
                                removeFromCart={removeItem}
                            />
                        ))}
                        <Divider />
                        <Stack sx={{ mt: 2 }} alignItems={{
                            sx: "center",
                            sm: "flex-start"
                        }}>
                            <Typography data-testid="total-cost">
                                Total Cost : {formatCurrency(totalCost)}
                            </Typography>
                        </Stack>


                        <Stack sx={{ mt: 2 }} alignItems={{
                            sx: "center",
                            sm: "flex-start"
                        }}>
                            <Button variant="contained" size="large" onClick={handlePayNow}>Pay Now</Button>
                        </Stack>
                    </>
                ) :
                    <EmptyCart />
            }
        </StyledBox>
    );
}

export default memo(FullCart);