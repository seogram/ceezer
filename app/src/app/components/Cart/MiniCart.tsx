
import { memo, useCallback, useState } from "react";
import { Typography, Box, Paper, Badge, Popover, IconButton } from "@mui/material";
import { styled } from '@mui/material/styles';
import ShoppingBagOutlinedIcon from '@mui/icons-material/ShoppingBagOutlined';
import { useCartActions } from "@/app/hooks";
import { CustomeLink } from "..";
import { CartItem as CartItemType } from "@/app/type";
import CartItem from "./CartItem";
import { formatCurrency } from "@/app/utils";
import EmptyCart from "./EmptyCart";

const StyledPaper = styled(Paper)(({ theme }) => ({
    padding: theme.spacing(2),
    textAlign: 'center',
    ...theme.typography.body2,
}));

type Props = {
    totalCost: number,
    cartItems: CartItemType[]
};

const MiniCart = ({ totalCost, cartItems }:Props) => {
    const { removeItem, addItem } = useCartActions();

    const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
    const isCartShown = Boolean(anchorEl);

    const handleCartOpen = useCallback((event: React.MouseEvent<HTMLElement>) => {
        setAnchorEl(event.currentTarget);
    }, []);

    const handleCartClose = useCallback(() => {
        setAnchorEl(null);
    }, []);

    return (
        <Box sx={{ pt: 2 }}>
            <Badge badgeContent={cartItems.length} color="secondary">
                <IconButton
                    size="large"
                    edge="end"
                    aria-label="show cart items"
                    aria-controls="mini-cart"
                    aria-haspopup="true"
                    onClick={handleCartOpen}
                    color="inherit"
                    data-testid="mini-cart-icon"
                >
                    <ShoppingBagOutlinedIcon />
                </IconButton>
            </Badge>
            <Popover
                open={isCartShown}
                anchorEl={anchorEl}
                onClose={handleCartClose}
                anchorOrigin={{
                    vertical: 'bottom',
                    horizontal: 'right',
                }}
            >
                <StyledPaper>
                    {cartItems.length < 1 ? (
                        <EmptyCart />
                    ) : (
                        <>
                            {cartItems.map(item => (
                                <CartItem
                                    key={item.id}
                                    item={item}
                                    addToCart={addItem}
                                    removeFromCart={removeItem}
                                />
                            ))}
                            <Typography data-testid="total-cost">Total Cost: {formatCurrency(totalCost)}</Typography>
                            <Box m={2}>
                                <CustomeLink href="/checkout">
                                    <Typography color="primary" variant="body1">Proceed to Checkout</Typography>
                                </CustomeLink>
                            </Box>
                        </>
                    )}
                </StyledPaper>
            </Popover>
        </Box>
    );
}

export default memo(MiniCart);