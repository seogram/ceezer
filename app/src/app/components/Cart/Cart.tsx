"use client";

import { useEffect, useMemo, useState } from "react";
import { useCart } from "@/app/hooks";
import MiniCart from "./MiniCart";
import FullCart from "./FullCart";

const Cart = ({ full = false }) => {
    const [totalCost, setTotalCost] = useState(0);
    const { cartItems } = useCart();

    useEffect(() => {
        const total = cartItems.reduce((sum, item) => sum + item.pricePerTon * item.volume, 0);
        setTotalCost(total);
    }, [cartItems]);

    const payload = useMemo(() => ({
        totalCost,
        cartItems
    }), [totalCost, cartItems]);

    if (full) {
        return <FullCart {...payload} />;
    }
    return <MiniCart {...payload} />;
};

export default Cart;