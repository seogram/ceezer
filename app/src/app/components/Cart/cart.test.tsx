import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import Cart from './Cart';
import { MockCartItemData } from "../Projects/data";
import { useCart } from "@/app/hooks";
import { CartItemsContext } from '@/app/context/CartItemsContext';

jest.mock("../../hooks/useCart");

jest.mock("next/navigation", () => ({
  useRouter: jest.fn()
}));
const mockSetCartItems = jest.fn();

const updatedCartAfterIncrease = [
  {
    "id": "1",
    "image": "https://ceezer-public-assets.s3.eu-central-1.amazonaws.com/project_type_sample_images/Fugitives/38bb530f5caf513be9f2a41f2d909f47-min.jpeg",
    "name": "EverGreen CarbonScape",
    "offeredVolume": 10,
    "pricePerTon": 650,
    "volume": 1.5
  },
  {
    "id": "3",
    "image": "https://ceezer-public-assets.s3.eu-central-1.amazonaws.com/project_type_sample_images/Afforestation+reforestation/marita-kavelashvili-ugnrXk1129g-unsplash-min.jpg",
    "name": "SustainaForest Carbon",
    "offeredVolume": 3, 
    "pricePerTon": 50,
    "volume": 2
  }
];

function setup() {
  const mockSetCartItems = jest.fn();
  const utils = render(
    <CartItemsContext.Provider value={{ cartItems: MockCartItemData.cartItems, setCartItems: mockSetCartItems }}>
      <Cart full />
    </CartItemsContext.Provider>
  );
  return {
    ...utils,
  };
}


describe('Cart component', () => {
  beforeEach(() => {
    (useCart as jest.Mock).mockReturnValue({
      cartItems: MockCartItemData.cartItems,
      setCartItems: mockSetCartItems
    });
  });
  afterEach(cleanup);

  it('renders without crashing', () => {
    const { asFragment } = setup();
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the total cost correctly', () => {
    const { getByTestId } = setup();
    expect(getByTestId("total-cost")).toHaveTextContent("Total Cost : â‚¬750.00");
  });

  it('correctly increases item volume and updates cart items', () => {
    const { getAllByTestId } = setup();

    fireEvent.click(getAllByTestId("increase-volume")[0]);
    expect(mockSetCartItems).toHaveBeenCalledWith(expect.arrayContaining(
      updatedCartAfterIncrease
    ));
  });
});


