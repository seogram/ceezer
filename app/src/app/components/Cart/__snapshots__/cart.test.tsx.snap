// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart component renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1b5cjn8"
    data-testid="cart"
  >
    <div
      class="MuiStack-root css-155h7j6-MuiStack-root"
    >
      <div
        class="MuiBox-root css-1rr4qq7"
      >
        <div
          class="MuiBox-root css-1em3tkh"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            EverGreen CarbonScape
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Price per ton: €650.00
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Volume(ton): 1
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Total Price: €650.00
          </p>
        </div>
        <div
          class="MuiStack-root css-kdc3n8-MuiStack-root"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
            data-testid="decrease-volume"
            tabindex="0"
            type="button"
          >
            -
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiBox-root css-0"
          >
            1
          </div>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
            data-testid="increase-volume"
            tabindex="0"
            type="button"
          >
            +
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiBox-root css-zjuml4"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeEnd MuiIconButton-sizeLarge css-1f5y6oa-MuiButtonBase-root-MuiIconButton-root"
              data-testid="remove-item"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DeleteOutlineIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM8 9h8v10H8zm7.5-5-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <img
          alt="EverGreen CarbonScape"
          data-nimg="1"
          decoding="async"
          height="160"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fceezer-public-assets.s3.eu-central-1.amazonaws.com%2Fproject_type_sample_images%2FFugitives%2F38bb530f5caf513be9f2a41f2d909f47-min.jpeg&w=384&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fceezer-public-assets.s3.eu-central-1.amazonaws.com%2Fproject_type_sample_images%2FFugitives%2F38bb530f5caf513be9f2a41f2d909f47-min.jpeg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fceezer-public-assets.s3.eu-central-1.amazonaws.com%2Fproject_type_sample_images%2FFugitives%2F38bb530f5caf513be9f2a41f2d909f47-min.jpeg&w=384&q=75 2x"
          style="color: transparent;"
          width="150"
        />
      </div>
      <div
        class="MuiBox-root css-130f8nx"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeEnd MuiIconButton-sizeLarge css-1f5y6oa-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="DeleteOutlineIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM8 9h8v10H8zm7.5-5-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiStack-root css-155h7j6-MuiStack-root"
    >
      <div
        class="MuiBox-root css-1rr4qq7"
      >
        <div
          class="MuiBox-root css-1em3tkh"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            SustainaForest Carbon
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Price per ton: €50.00
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Volume(ton): 2
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Total Price: €100.00
          </p>
        </div>
        <div
          class="MuiStack-root css-kdc3n8-MuiStack-root"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
            data-testid="decrease-volume"
            tabindex="0"
            type="button"
          >
            -
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiBox-root css-0"
          >
            2
          </div>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
            data-testid="increase-volume"
            tabindex="0"
            type="button"
          >
            +
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiBox-root css-zjuml4"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeEnd MuiIconButton-sizeLarge css-1f5y6oa-MuiButtonBase-root-MuiIconButton-root"
              data-testid="remove-item"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DeleteOutlineIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM8 9h8v10H8zm7.5-5-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <img
          alt="SustainaForest Carbon"
          data-nimg="1"
          decoding="async"
          height="160"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fceezer-public-assets.s3.eu-central-1.amazonaws.com%2Fproject_type_sample_images%2FAfforestation%2Breforestation%2Fmarita-kavelashvili-ugnrXk1129g-unsplash-min.jpg&w=384&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fceezer-public-assets.s3.eu-central-1.amazonaws.com%2Fproject_type_sample_images%2FAfforestation%2Breforestation%2Fmarita-kavelashvili-ugnrXk1129g-unsplash-min.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fceezer-public-assets.s3.eu-central-1.amazonaws.com%2Fproject_type_sample_images%2FAfforestation%2Breforestation%2Fmarita-kavelashvili-ugnrXk1129g-unsplash-min.jpg&w=384&q=75 2x"
          style="color: transparent;"
          width="150"
        />
      </div>
      <div
        class="MuiBox-root css-130f8nx"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeEnd MuiIconButton-sizeLarge css-1f5y6oa-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="DeleteOutlineIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM8 9h8v10H8zm7.5-5-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <div
      class="MuiStack-root css-19qajtz-MuiStack-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        data-testid="total-cost"
      >
        Total Cost : €750.00
      </p>
    </div>
    <div
      class="MuiStack-root css-19qajtz-MuiStack-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary css-jh47zj-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Pay Now
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
